### DISCUSSION TITLE
issue connecting to payment (Android)

### DISCUSSION THREAD

**Discussion by claudiaTonak** [hyochan/react-native-iap#2276] [General]
I try to request a subscription or a single purchase, Android.
Both calls got the same error message:
 {"message":"Google is indicating that we have some issue connecting to payment.","debugMessage":"Please ensure the specific App version has been published.","code":"E_DEVELOPER_ERROR","responseCode":5} 

react-native-iap version 12.8.1
react-native 0.71.3

Calls:

        await IAP.requestSubscription({
          sku, 
          ...(offerToken && {
            subscriptionOffers: [{ sku, offerToken }],
          }),
        });

      await IAP.requestPurchase(
          Platform.select({
            ios: { sku },
            android: { skus: [sku] },
          })
        );

the sku and offerToken are correct.

I also tried:

        await IAP.requestPurchase(
          Platform.select({
            ios: { sku },
            android: {
              skus: [sku],
              obfuscatedProfileIdAndroid: editionName,
              obfuscatedAccountIdAndroid: curPublDate,
            },
          })
        );

without success, same failure.

With react-native-iap 7.5.6 and react-native 17.0.2 the requestPurchase and requestSubscription calls are working just fine.

          await IAP.requestSubscription(sku, false, undefined, undefined, editionName, curPublDate);
          await IAP.requestPurchase(sku, false, editionName, curPublDate);

Any ideas? Can anybody help me please?

**Comment by andresesfm** [comment]
Look at the migration docs. You should be passing an object to those methods

**Reply by claudiaTonak** [reply]
This is not the solution. I passed an object. Please take a closer look.
Android: { skus: [sku],
          obfuscatedProfileIdAndroid: editionName,
          obfuscatedAccountIdAndroid: curPublDate,
        }

**Reply by sheikgulf13** [reply]
did you find a solution? i have the same issue.

**Comment by thesheikhwasim** [comment]
Release your app bundle to internal testing and you will see things working.
